File: spatial-index/src/main/java/lva/spatialindex/storage/StorageSpace.java
package lva.spatialindex.storage;

/**
 * @author vlitvinenko
 */
public interface StorageSpace {
    byte[] readBytes(long pos, int size);
    void readBytes(long pos, byte[] buff);
    void writeBytes(long pos, byte[] data);
    long allocate(long sizeOf);
    long getSize();
    long getCapacity();
    void clear();
}

File: spatial-index/src/main/java/lva/spatialindex/storage/StorageSpace.kt
package lva.spatialindex.storage

/**
 * @author vlitvinenko
 */
interface StorageSpace {
    val size: Long
    val capacity: Long
    fun readBytes(pos: Long, size: Int): ByteArray
    fun readBytes(pos: Long, buff: ByteArray)
    fun writeBytes(pos: Long, data: ByteArray)
    fun allocate(sizeOf: Long): Long
    fun clear()
}